sonar-csharp-plugin/src/main/resources/documentation.md

```diff
@@ -16,12 +16,26 @@ Discover and update the C#-specific [properties](/analysis/analysis-parameters/)
 
 To analyze tool-generated code (e.g. WCF code generated by `SvcUtil.exe`, protobuf code generated by `protoc`, Swagger client code generated by `NSwag`) for a specific C# project, enable the "Analyze generated code" setting inside **Project > Administration > General Settings > C#**. By default, tool-generated code files are skipped from analysis.
 
+Generated code detection is based on file name, special comments and attributes. Currently recognized values are in [GeneratedCodeRecognizer.cs](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Common/Helpers/GeneratedCodeRecognizer.cs).
+
 ## Scanner compatibility
 
 To analyze C# code, you need to use the [SonarScanner for .NET](/analysis/scan/sonarscanner-for-msbuild/) version 4.x or newer.
 
+## Exclusions
+
+Files excluded from analysis should be set in a project configuration. This exclusion still performs the analysis on the file and filters final results.
```

Comment:
```suggestion
The files which are to be excluded from analysis should be set in the project configuration. The files are still going to be analyzed during the MSBuild compilation, the exclusion happening as a filter before sending the results to the server.
```

Reply:
Are you sure about the `which are to be`? That one sounds very complicated/future-conditional-ish.

Reply:
`the exclusion happening as a` - isn't there a verb missing? `the exclusion is happening as a`

Reply:
The `before sending the results to the server` is providing too much technical details IMO.

Reply:
the idea is that 
> This exclusion still performs the analysis on the file and filters final results.
Is not clear at all, because:
- we say "the exclusion performs the analysis", which isn't the case (the exclusion doesn't perform anything)
- either way, our users don't know when the analysis is done, that it is done during the build etc

AND I agree with you it's too much detail :). So a new version:

> The files which are to be excluded from analysis should be set in the project configuration. The files are still going to be analyzed during the MSBuild compilation. The configuration of exclusions is used to filter the analysis results.